File: app/src/main/java/com/lumiyaviewer/lumiya/render/spatial/MyAvatarTreeNode.java
package com.lumiyaviewer.lumiya.render.spatial;

import com.lumiyaviewer.lumiya.utils.InlineList;

/* loaded from: classes.dex */
public class MyAvatarTreeNode extends InlineList<DrawListEntry> {
    private final SpatialTree spatialTree;

    public MyAvatarTreeNode(SpatialTree spatialTree) {
        this.spatialTree = spatialTree;
    }

    public void addDrawables(DrawList drawList) {
        for (DrawListEntry first = getFirst(); first != null; first = first.getNext()) {
            first.addToDrawList(drawList);
        }
    }

    @Override // com.lumiyaviewer.lumiya.utils.InlineList
    public void addEntry(DrawListEntry drawListEntry) {
        super.addEntry((MyAvatarTreeNode) drawListEntry);
        this.spatialTree.setDrawListChanged();
    }

    @Override // com.lumiyaviewer.lumiya.utils.InlineList
    public void removeEntry(DrawListEntry drawListEntry) {
        super.removeEntry((MyAvatarTreeNode) drawListEntry);
        this.spatialTree.setDrawListChanged();
    }

    @Override // com.lumiyaviewer.lumiya.utils.InlineList
    public void requestEntryRemoval(DrawListEntry drawListEntry) {
        this.spatialTree.spatialObjectIndex.requestEntryRemoval(drawListEntry);
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/render/spatial/MyAvatarTreeNode.kt
package com.lumiyaviewer.lumiya.render.spatial

import com.lumiyaviewer.lumiya.utils.InlineList

class MyAvatarTreeNode internal constructor(private val spatialTree: SpatialTree) : InlineList<DrawListEntry>() {

    fun addDrawables(drawList: DrawList) {
        var entry = first
        while (entry != null) {
            entry.addToDrawList(drawList)
            entry = entry.getNext()
        }
    }

    override fun addEntry(drawListEntry: DrawListEntry) {
        super.addEntry(drawListEntry)
        spatialTree.setDrawListChanged()
    }

    override fun removeEntry(drawListEntry: DrawListEntry) {
        super.removeEntry(drawListEntry)
        spatialTree.setDrawListChanged()
    }

    override fun requestEntryRemoval(drawListEntry: DrawListEntry) {
        spatialTree.spatialObjectIndex.requestEntryRemoval(drawListEntry)
    }
}
